rpc/src/rpc.rs

```diff
@@ -3468,9 +3468,15 @@ pub mod rpc_full {
         ) -> Result<String> {
             debug!("send_transaction rpc request received");
             let config = config.unwrap_or_default();
-            let encoding = config.encoding.unwrap_or(UiTransactionEncoding::Base58);
+            let tx_encoding = config.encoding.unwrap_or(UiTransactionEncoding::Base58);
+            let binary_encoding = tx_encoding.into_binary_encoding().ok_or_else(|| {
```

Comment:
Since the binary variants are labeled the same and get sent over the wire in json, I think that we could actually get away with updating the types in the config structs (`RpcSendTransactionConfig` and `RpcSimulateTransactionConfig`), and forgo this conversion.
It would mean rust clients using those config structs directly would need to update the type then they update, but there should be no urgency to update. Wdyt?

Reply:
It would be a backwards incompatible change in the Solana client api so I was thinking it can come later. How were you imagining the transition?

Reply:
Ah, yeah, I suppose the RPC error change would technically be backwards incompatible. Is this your main concern?

On the client side, I was actually thinking just break the `simulate_transaction_with_config` and `send_transaction_with_config` (etc) methods, since old and new versions would both be compatible with old and new RPCs. But I suppose the less breaky way to do it would be to add new methods and deprecate the existing ones (which could do the UiTransactionEncoding to TransactionBinaryEncoding conversion under the hood).

Anyway, feel free to merge this as-is in the meantime.

Reply:
I was talking about breaking the solana-client rust API actually. Technically should happen in a major version update, right? I think adding new methods and deprecating the old ones is the way to go.